Client/Frontend/Browser/SearchEnginesJSON.swift

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public class SearchEnginesJSON {
+    private let json: JSON
+
+    public init(_ jsonString: String) {
+        self.json = JSON.parse(jsonString)
+    }
+
+    public init(_ json: JSON) {
+        self.json = json
+    }
+
+    public func visibleDefaultEngines(possibilities: [String], region: String) -> [String] {
+        var engineNames: [JSON]? = nil
+        for possibility in possibilities {
+            if (json["locales"][possibility].asDictionary != nil) {
+                engineNames = json["locales"][possibility][region]["visibleDefaultEngines"].asArray
+                if (engineNames == nil) {
+                    engineNames = json["locales"][possibility]["default"]["visibleDefaultEngines"].asArray
+                }
+            }
```

Comment:
That's a good question. I'm not sure. If we for some reason have en-US with a default and then en with a US specific value, I think we'd want to take that. 

I think we actually want to break when we get a non default match...

Then the last thing we would take would be the last default we found. I think.